internal/providercache/dir_modify.go

```diff
@@ -67,5 +67,9 @@ func (d *Dir) LinkFromOtherCache(entry *CachedProvider) error {
 	// We re-use the process of installing from a local directory here, because
 	// the two operations are fundamentally the same: symlink if possible,
 	// deep-copy otherwise.
-	return installFromLocalDir(context.TODO(), currentPath, newPath)
+	meta := getproviders.PackageMeta{
+		Location: getproviders.PackageLocalDir(currentPath),
```

Comment:
This `PackageMeta` with only `Location` set feels a little awkward in that `installFromLocalDir` function must now be aware that it can potentially receive an incomplete/invalid object :confused: but I must admit I don't really have a better idea... this seems like a pragmatic approach for now, and it's an unexported API so not something we're blocked from changing if we want to later.

Reply:
Agreed that it's awkward, and I also can't think of a better solution right now. I think there's a bit of an improvement in cec607f38, where we now use the same approach as this `PackageMeta` construction logic from `getproviders.SearchLocalDirectory`: https://github.com/hashicorp/terraform/blob/76f583ac9fdbfca9f23257753094f1a1bef5f686/internal/getproviders/filesystem_search.go#L102-L118